app/assets/src/components/views/SampleUploadFlow/UploadSampleStep.jsx

```diff
@@ -370,52 +286,123 @@ class UploadSampleStep extends React.Component {
     ]);
 
     this.setState({
-      validatingSamples: false,
-      projects: concat(project, this.state.projects),
       selectedProject: project,
-      createProjectOpen: false,
+      validatingSamples: false,
       localSamples: newLocalSamples,
       remoteSamples: newRemoteSamples,
       basespaceSamples: newBasespaceSamples,
-      sampleNamesToFiles: newSampleNamesToFiles,
     });
 
-    logAnalyticsEvent("UploadSampleStep_project_created", {
+    logAnalyticsEvent("UploadSampleStep_project-selector_changed", {
       localSamples: newLocalSamples.length,
       remoteSamples: newRemoteSamples.length,
+      basespaceSamples: newBasespaceSamples.length,
       ...this.getAnalyticsContext(),
     });
   };
 
-  handleTabChange = tab => {
-    this.props.onDirty();
-    this.setState({ currentTab: tab });
-    logAnalyticsEvent("UploadSampleStep_tab_changed", {
-      tab,
+  handleNoProject = () => {
+    this.setState({
+      showNoProjectError: true,
+    });
+  };
+
+  openCreateProject = () => {
+    this.setState({
+      createProjectOpen: true,
+    });
+  };
+
+  closeCreateProject = () => {
+    this.setState({
+      createProjectOpen: false,
     });
   };
 
+  //*** Sample-related functions ***
+
+  // Functions to get the state key by sample type, e.g. this.state.localSamples, this.state.basespaceSamples
+  getSelectedSampleIdsKey = sampleType => `${sampleType}SelectedSampleIds`;
+  getSamplesKey = sampleType => `${sampleType}Samples`;
+
+  getSelectedSampleIds = sampleType => {
+    return this.state[this.getSelectedSampleIdsKey(sampleType)];
+  };
+
+  getSelectedSamples = sampleType => {
+    const selectedSampleIds = this.getSelectedSampleIds(sampleType);
+    const samplesKey = this.getSamplesKey(sampleType);
+    return filter(
+      sample => selectedSampleIds.has(sample._selectId),
+      this.state[samplesKey]
+    );
+  };
+
+  // Get fields for display in the SampleUploadTable.
+  getSampleDataForUploadTable = sampleType => {
+    if (sampleType === "basespace") {
+      return this.state.basespaceSamples;
+    }
+
+    if (sampleType === "remote" || sampleType === "local") {
+      const samplesKey = this.getSamplesKey(sampleType);
+      const samples = this.state[samplesKey];
+      return samples.map(sample => ({
+        ...pick(["name", SELECT_ID_KEY], sample),
+        file_names: map(
+          file => file.name || file.source,
+          sample.input_files_attributes
+        ).sort(),
+      }));
+    }
+  };
+
+  areAllSamplesSelected = sampleType => {
```

Comment:
This would have been taken care of by the table... is this an issue due to the fact you were trying to use the same table for all upload types?

Reply:
I missed that this was in the table. Thanks!